### ISSUE TITLE
Issue with handling oData.NextLink in expanded entities?

### ISSUE THREAD

**Issue by somefella123** [object/Simple.OData.Client#279]
Hi,

I'm trying to use your library with the CRM 2016 oData v4 endpoint, described on [this page](https://msdn.microsoft.com/en-us/library/gg334767.aspx) & have hit a blocker.

Specifically in the scenario described if you search that page for the text "_Retrieve related entities by expanding collection-valued navigation properties_"

I can't work out how to follow / resolve oData NextLinks embedded in expanded entities... The expanded entitiy simply isn't present at all in the DynamicODataEntry objects that come back from my query (clearly I'm using the dynamic api). The http response for my query is in the same format as the response below taken from msdn - it's the Account_Tasks section which causes the issue.

Should the library be handling this kind of server response okay?:

(partial response snippet):

```
HTTP/1.1 200 OK
Content-Type: application/json; odata.metadata=minimal
OData-Version: 4.0

{
  "@odata.context":"[Organization URI]/api/data/v8.1/$metadata#accounts(name,Account_Tasks,Account_Tasks(subject,scheduledstart))","value":[
    {
      "@odata.etag":"W/\"513475\"","name":"Fourth Coffee (sample)","accountid":"36dbf27c-8efb-e511-80d2-00155db07c77","Account_Tasks":[

      ],"Account_Tasks@odata.nextLink":"[Organization URI]/api/data/v8.1/accounts(36dbf27c-8efb-e511-80d2-00155db07c77)/Account_Tasks?$select%20=%20subject,%20scheduledstart"
    },{
```

**Comment by object** [comment]
Hi,

Can you check these tests (look at FindAllPeople that uses OData annotations) and see if it helps:
https://github.com/object/Simple.OData.Client/blob/master/Simple.OData.Client.IntegrationTests/TripPinTests.cs

**Comment by somefella123** [comment]
Hi Vagif - unfortunately those don't help (but I agree the tests are an awesome source of tutorials). In my scenario the NextLink is on one of the expanded entities and represents a 1:m navigation, rather than being used for "paging" multiple results.

In other words, I receive all ~50 top level entities (so annotations.NextPageLink is null as all entities were returned), but some of the detail within each entity is missing: the "nextlink" that was passed back from the service instead of the expanded entity's details has been lost somewhere.

**Comment by object** [comment]
I see. You don't have by chance public endpoint that I can access? I would like to investigate this thoroughly.

**Comment by somefella123** [comment]
I've set up a trial CRM Online instance - you can access this to manipulate the odata endpoint and see the results above via a browser - it might be possible to also access this via OAUTH but I've not had time to investigate/set this up properly unfortunately... let me know what address to send you login details to?...

**Comment by object** [comment]
Please send the login details to vagif.abilov at gmail.

**Comment by object** [comment]
Resolved in 4.22.
